chore: improve clarity and consistency of doc descriptions

kylecarbs opened the pull request:
Ran this through ChatGPT with a few iterations to improve the clarity of each description.

Soon our docs will have an `og:image` and a social image auto-generated for each page:

![image](https://github.com/coder/coder/assets/7122116/d8876145-045e-4e67-b7d7-707091af0d64)

stirby commented on docs/manifest.json:
```diff
@@ -3,268 +3,270 @@
         {
           "title": "Icons",
-          "description": "Coder includes icons for popular cloud providers and programming languages for you to use",
+          "description": "Learn about the icons C
+
+oder includes for popular cloud providers and programming languages.",
```
What's this?

kylecarbs replied:
Awkward

kylecarbs replied:
Fixed

stirby commented on docs/manifest.json:
```diff
@@ -277,266 +279,269 @@
     {
       "title": "IDEs",
-      "description": "Learn how to use your IDE of choice with Coder",
+      "description": "Learn how to use your IDE of choice with Coder.",
```
I don't think we should use periods here.

stirby commented on docs/manifest.json:
```diff
@@ -1068,72 +1073,72 @@
     {
       "title": "Guides",
-      "description": "Employee-authored tutorials",
+      "description": "Learn from employee-authored tutorials.",
```
It seems like most of these just add "Learn from" or "Learn about" which just elongates the descriptions?

kylecarbs replied:
These are meta tags used for SEO. Would you rather "Learn from" be prefixed in the Google description? I certainly think it adds clarity.

kylecarbs committed:
chore: improve clarity and consistency of doc descriptions

Ran this through ChatGPT with a few iterations to improve
the clarity of each description.
